src/mutation.rs

```diff
@@ -0,0 +1,225 @@
+use std::any::type_name;
+use std::collections::BTreeSet;
+
+use async_graphql::{Context, Error, Object, Result};
+use bson::Bson;
+use bson::Uuid;
+use futures::TryStreamExt;
+use mongodb::{
+    bson::{doc, DateTime},
+    Collection, Database,
+};
+use serde::Deserialize;
+
+use crate::authentication::authenticate_user;
+use crate::foreign_types::Discount;
+use crate::foreign_types::ProductItem;
+use crate::foreign_types::ProductVariantVersion;
+use crate::foreign_types::ShipmentMethod;
+use crate::foreign_types::TaxRateVersion;
+use crate::mutation_input_structs::CreateOrderInput;
+use crate::mutation_input_structs::OrderItemInput;
+use crate::order::OrderStatus;
+use crate::order_item::OrderItem;
+use crate::query::query_object;
+use crate::user::User;
+use crate::{order::Order, query::query_order};
+
+/// Describes GraphQL order mutations.
+pub struct Mutation;
+
+#[Object]
+impl Mutation {
+    /// Creates an order with `OrderStatus::Pending`.
+    async fn create_order<'a>(
+        &self,
+        ctx: &Context<'a>,
+        #[graphql(desc = "CreateOrderInput")] input: CreateOrderInput,
```

Comment:
So, I'm no going to be really annoying, but this does not really match the flow we discussed in the beginning of the project.
We decided that to order stuff, you need to put stuff in the shoppingcart, and then you can order shoppingcart items.
So you should get all the input from the created / updated events of your shoppingcart service.
Here, you then only get the ids of which shoppingcart items the user currently actually wants to buy.
You also need to throw an event after the place_order thing to remove the ordered items from the shoppingcart
Alternatively, you could already remove them here, however if placing the order fails, they would need to be recreated which is probably more work